report/sql_tasks/tasks/report/create_from_scratch/04_hubspot_write/01_company_property_update/01_create_view.sql

```diff
@@ -0,0 +1,76 @@
+DROP VIEW IF EXISTS hubspot.company_property_update CASCADE;
+
+-- This view recreates the properties that we update on Hubspot companies
+CREATE VIEW hubspot.company_property_update AS (
+    WITH
+        metrics AS (
+            SELECT
+                column1 AS metric_name,
+                column2 AS role,
+                column3 AS timescale,
+                column4 AS offset
+            FROM (
+                VALUES
+                    ('lms_teachers_last_semester', 'teacher', 'semester', 1),
+                    ('lms_teachers_this_semester', 'teacher', 'semester', 0),
+                    ('lms_users_last_academic_year', 'user', 'academic_year', 1),
+                    ('lms_users_this_academic_year', 'user', 'academic_year', 0),
+                    ('lms_users_last_semester', 'user', 'semester', 1),
+                    ('lms_users_this_semester', 'user', 'semester', 0),
+                    ('lms_users_all_time', 'user', 'all_time', 0)
+            ) AS data
+        ),
+
+        values_in_rows AS (
+            SELECT
+                activity.organization_id,
+                metrics.metric_name,
+                activity.billable as count
+            FROM metrics
+
+            -- Get the correct time period to join on to
+            LEFT JOIN LATERAL (
+                SELECT DISTINCT(period)
+                FROM organization_activity
+                WHERE timescale::text = metrics.timescale
+                ORDER BY period DESC
+                LIMIT 1
+                OFFSET metrics.offset
+            ) metric_period ON TRUE
+
+            -- Get the correct metric rows
+            LEFT OUTER JOIN organization_activity ON
+                organization_activity.timescale::text = metrics.timescale
+                AND organization_activity.role::text = metrics.role
+                AND organization_activity.period = metric_period.period
```

Comment:
This is ugly, but less ugly than joining together separate queries for each of the metrics.